zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/SpotRestorationTool.java

```diff
@@ -0,0 +1,276 @@
+package org.apache.zookeeper.server.backup;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import org.apache.jute.Record;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Stat;
+import org.apache.zookeeper.server.DataNode;
+import org.apache.zookeeper.server.DataTree;
+import org.apache.zookeeper.server.persistence.FileTxnSnapLog;
+import org.apache.zookeeper.txn.TxnDigest;
+import org.apache.zookeeper.txn.TxnHeader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a tool for automatic restoration of ZK data based on a zk node path using a collection of
+ * snapshot and transaction logs.
+ */
+public class SpotRestorationTool {
+  private static final Logger LOG = LoggerFactory.getLogger(SpotRestorationTool.class);
+
+  File dataDir;
+  ZooKeeper zk;
+  FileTxnSnapLog snapLog;
+  boolean restoreRecursively;
+  String targetZNodePath;
+  List<String> messages;
+
+  /**
+   * Constructor
+   * @param dataDir The directory contains snapshot and transaction log files to be used for restoration
```

Comment:
Nit: "the directory/path that contains snapshot and transaction..."

Let's also clarify if these are backup files or fully-restored snapshot/TxLog files?